File: nandincpp/Count Number of Pairs With Absolute Difference K.java
class Solution {
    public int countKDifference(int[] a, int k) 
    {
        HashMap<Integer, Integer> m = new HashMap<Integer, Integer>();
        int ans = 0;
        
        for(int i=0;i<a.length;i++)
        {
            if(m.containsKey(a[i]+k))
                ans += m.get(a[i]+k);
            if(m.containsKey(a[i]-k))
                ans += m.get(a[i]-k);
            
            m.put(a[i], m.getOrDefault(a[i], 0) + 1);
        }
        
        return ans;
    }
}

File: nandincpp/Count Number of Pairs With Absolute Difference K.cpp
class Solution {
public:
    int countKDifference(vector<int>& a, int k) 
    {
        unordered_map<int, int> m;
        int ans = 0;
        
        for(int i=0;i<a.size();i++)
        {
            if(m.count(a[i]+k))
                ans += m[a[i]+k];
            if(m.count(a[i]-k))
                ans += m[a[i]-k];
            
            m[a[i]]++;
        }
        return ans;
    }
};